src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,225 @@
+    inputs.leftSupplyCurrentAmps = leftSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSetpoint = leftSetpointStatusSignal.getValueAsDouble();
+
+    inputs.rightVelocityRPS = rightVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.rightPositionMeters = rightPositionMetersStatusSignal.getValueAsDouble();
+    inputs.rightStatorCurrentAmps = rightStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSupplyCurrentAmps = rightSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSetpoint = rightSetpointStatusSignal.getValueAsDouble();
+
+    // inputs.closedLoopError = leftMotor.getClosedLoopError().getValue();
+    // inputs.power =
+    // inputs.controlMode = leftMotor.getControlMode().toString();
+
+    // update configuration if tunables have changed
+    if (kP.hasChanged()
+        || kI.hasChanged()
+        || kD.hasChanged()
+        || kS.hasChanged()
+        || kA.hasChanged()
+        || kV.hasChanged()
+        || kVExpo.hasChanged()
+        || kAExpo.hasChanged()) {
+      TalonFXConfiguration config = new TalonFXConfiguration();
+      MotionMagicConfigs motionMagicConfigs = new MotionMagicConfigs();
+
+      this.leftMotor.getConfigurator().refresh(config);
+      config.Slot0.kP = kP.get();
+      config.Slot0.kI = kI.get();
+      config.Slot0.kD = kD.get();
+      config.Slot0.kS = kS.get();
+      config.Slot0.kV = kV.get();
+      config.Slot0.kA = kA.get();
+      motionMagicConfigs.MotionMagicExpo_kV = kVExpo.get();
+      motionMagicConfigs.MotionMagicExpo_kA = kAExpo.get();
+
+      this.leftMotor.getConfigurator().apply(config);
+      this.rightMotor.getConfigurator().apply(config);
+    }
+  }
+
+  /**
+   * Set the leftMotor position to the specified value in degrees.
+   *
+   * @param position the position to set the leftMotor to in meters
+   * @param arbitraryFeedForward the arbitrary feed forward as a percentage of maximum power
+   */
+  @Override
+  public void setLeftMotorPosition(double position) {
+    leftMotor.setControl(
+        leftPositionCurrentRequest
+            .withPosition(Conversions.degreesToFalconRotations(position, GEAR_RATIO))); // FIXME: Feed forward should be acounted for already.
+  }
+
+  @Override
+  public void setRightMotorPosition(double position) {
+    this.rightMotor.setControl(
+        rightPositionCurrentRequest
+            .withPosition(Conversions.degreesToFalconRotations(position, GEAR_RATIO)));
+  }
+
+  @Override
+  public void setLeftMotorPower(double current) {
+    leftMotor.setControl(leftSupplyAmpsRequest.withOutput(current));
+  }
+
+  @Override
+  public void setRightMotorPower(double current) {
+    rightMotor.setControl(rightSupplyAmpsRequest.withOutput(current));
+  }
+
+  @Override
+  public void setPositionZero() {
+    leftEncoder.setPosition(0);
+    rightEncoder.setPosition(0);
+  }
+
+  private void configMotors(TalonFX leftMotor, TalonFX rightMotor) {
```

Comment:
You probably need to configure the left and right motors slightly differently in terms of inverting. So, I recommend that you change the parameters to be canID and isInverted. (The current parameters aren't actually used....) Implement this method to configure a single motor based on these parameters and then invoke this method twice in the constructor.

Reply:
and a third parameter as a description for the registerHardware invocation